src/libostree/ostree-sysroot-deploy.c

```diff
@@ -973,17 +978,88 @@ checksum_from_kernel_src (const char   *name,
 }
 
 static gboolean
-syncfs_dir_at (int            dfd,
-               const char    *path,
-               GCancellable  *cancellable,
-               GError       **error)
+fsfreeze_thaw_cycle (int            rootfs_dfd,
```

Comment:
Can we have a small comment here with some of the rationale from the commit msg? E.g. why we're doing `FIFREEZE/FITHAW` instead of just `syncfs`.